src/fundus/scraping/pipeline.py

```diff
@@ -53,9 +169,44 @@ def crawl(
         restrict_sources_to: Optional[List[Type[URLSource]]] = None,
         error_handling: Literal["suppress", "catch", "raise"] = "suppress",
         only_complete: Union[bool, ExtractionFilter] = True,
-        batch_size: int = 10,
+        delay: Optional[Union[float, Delay]] = None,
+        url_filter: Optional[URLFilter] = None,
+        only_unique: bool = True,
     ) -> Iterator[Article]:
+        """Yields articles from initialized publishers
+
+        Args:
+            max_articles (Optional[int]): Number of articles to retrieve. If there are fewer articles
+                than max_articles the Iterator will stop before max_articles. If None, all retrievable
+                articles are returned. Defaults to None.
+            restrict_sources_to (Optional[List[Literal["rss", "sitemap", "news"]]]): Let's you restrict
+                sources defined in the publisher specs. If set, only articles from given source types
+                will be yielded.
+            error_handling (Literal["suppress", "catch", "raise"]): Define how to handle errors
+                encountered during extraction. If set to "suppress", all errors will be skipped, either
+                with None values for respective attributes in the extraction or by skipping entire articles.
+                If set to "catch", errors will be caught as attribute values or, if an entire article fails,
+                through Article.exception. If set to "raise" all errors encountered during extraction will
+                be raised. Defaults to "suppress".
+            only_complete (Union[bool, ExtractionFilter]): Set extraction filters. If False, all articles
+                will be yielded, if True, only complete ones. Defaults to True. See the docs for more
+                information about ExtractionFilter.
+            delay (Optional[Union[float, Delay]]): Set a delay time in seconds to be used between article
+                batches. You can set a delay directly using float or any callable satisfying the Delay
+                protocol. If set to None, no delay will be used between batches. See Delay for more
+                information. Defaults to None.
+            url_filter (Optional[URLFilter]): A callable object satisfying the URLFilter protocol to skip
+                urls during download. This filter applies on both requested and responded URL. Defaults to None.
+            only_unique (bool): If set to True only the articles yielded will be unique on the responded URL.
```

Comment:
This sentence has some gramatical issues.

Reply:
Agree.